src/asm/lexer.c

```diff
@@ -1451,13 +1494,34 @@ static void readString(void)
 				shiftChars(1);
 				break;
 
+			/* Line continuation */
 			case ' ':
 			case '\r':
 			case '\n':
 				shiftChars(1); /* Shift the backslash */
 				readLineContinuation();
 				continue;
 
+			/* Macro arg */
+			case '@':
+			case '#':
+			case '0':
+			case '1':
+			case '2':
+			case '3':
+			case '4':
+			case '5':
+			case '6':
+			case '7':
+			case '8':
+			case '9':
+				shiftChars(2);
+				char const *str = readMacroArg(c);
+
+				if (str)
```

Comment:
Wait, isn't `readMacroArg`'s return value guaranteed to be non-NULL?

Besides, if it were NULL, shouldn't a diagnostic be printed? This is currently not relevant, but it's worth discussing—I'd be interested in making undefined macro args non-fatal, as less fatal errors = good.

Reply:
Currently in this PR, `readMacroArg` does a `fatalerror` ("Macro argument not defined") if it would otherwise return `NULL`. The `readMacroArg` call in `peek` assumes it's not null and just checks for an empty string (potentially returned by `macro_GetAllArgs` for `\#`) but this `readMacroArg` call in `readString` has the redundant null check. Should it be removed, or should `readMacroArg` just do a regular `error` and return `NULL`, in which case `peek` and `readString` would simply not do `beginExpansion` or `appendMacroArg` for that?

Reply:
For now, it should behave in accordance with current behavior, i.e. not perform any extra NULL checks. It would be added back, and properly dealt with, in the change that would non-`fatal` the `error` in `readMacroArg`.

Reply:
Okay.